subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/junit/result/AggregateTestResultsProvider.java

```diff
@@ -99,20 +99,33 @@ void addTestClassResult(TestClassResult delegate) {
     }
 
     @Override
-    public boolean hasOutput(long id, final TestOutputEvent.Destination destination) {
+    public boolean hasOutput(long classId, final TestOutputEvent.Destination destination) {
         return Iterables.any(
-                classOutputProviders.get(id),
-                new Predicate<DelegateProvider>() {
-                    @Override
-                    public boolean apply(DelegateProvider delegateProvider) {
-                        return delegateProvider.provider.hasOutput(delegateProvider.id, destination);
-                    }
-                });
+            classOutputProviders.get(classId),
```

Comment:
IIUC, the test result aggregation and test report generation happen at daemon's side, so can we use Java 8 Stream here?

Reply:
This is a Java 6 module: https://github.com/gradle/gradle/blob/30fd0fe882753347593d653e14a36ca7e06ce16e/subprojects/testing-base/build.gradle.kts#L22-L21